app/forms/Choice.scala

```diff
@@ -18,32 +18,40 @@ package forms
 
 import forms.Mapping.requiredRadio
 import play.api.data.{Form, Forms, Mapping}
-import play.api.libs.json.Json
-import utils.validators.forms.FieldValidator.isContainedIn
+import play.api.libs.json.{Format, JsError, JsResult, JsString, JsSuccess, JsValue, Reads}
 
-case class Choice(value: String)
+sealed abstract class Choice(val value: String)
 
 object Choice {
-  implicit val format = Json.format[Choice]
 
   val choiceId = "Choice"
 
-  import AllowedChoiceValues._
-  private val correctChoices = Set(Arrival, Departure, AssociateDUCR, DisassociateDUCR, ShutMucr, Submissions)
-
-  val choiceMapping: Mapping[Choice] = Forms.mapping(
-    "choice" -> requiredRadio("choicePage.input.error.empty")
-      .verifying("choicePage.input.error.incorrectValue", isContainedIn(correctChoices))
-  )(Choice.apply)(Choice.unapply)
+  val choiceMapping: Mapping[Choice] =
+    Forms.mapping("choice" -> requiredRadio("choicePage.input.error.empty"))(Choice.apply)(Choice.unapply)
```

Comment:
Here where we would try apply non existing choice we would get Exception from `form.bind` - please add validation step with some error. 

From perspective of UX - it better to have rerendered form with value with error then having "Internal Server Error" page

Reply:
Fixed